go/libraries/utils/argparser/parser.go

```diff
@@ -49,15 +49,31 @@ func ValidatorFromStrList(paramName string, validStrList []string) ValidationFun
 }
 
 type ArgParser struct {
+	Name              string
+	MaxArgs           int
+	TooManyArgsError  func(receivedArgs []string) error
```

Comment:
(minor/nit) I'd suggest naming this something like `TooManyArgsErrorFunc` – people are going to see `FooError` and naturally think "This thing is an error", so adding `Func` at the end seems a little more self-documenting.

Reply:
Fixed with https://github.com/dolthub/dolt/pull/5771